### PR TITLE
Refactor to use Function

### LINKED ISSUE #13: Refactor as Function Bean
https://github.com/spring-cloud-stream-app-starters/python/issues/18 depends on this

### PR DESCRIPTION
**Pull request by dturanski**
Fixes #13

### CODE REVIEW

File: spring-cloud-starter-stream-processor-httpclient/src/main/java/org/springframework/cloud/stream/app/httpclient/processor/HttpclientProcessorFunctionConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+					body = message.getPayload();
+				}
+				URI uri = new URI(url);
+				RequestEntity<?> request = new RequestEntity<>(body, headers, method, uri);
+				ResponseEntity<?> response = restTemplate.exchange(request, responseType);
+				return properties.getReplyExpression().getValue(response);
+			}
+			catch (Exception e) {
+				LOG.warn("Error in HTTP request", e);
+				return null;
```
Comment (artembilan) [inline spring-cloud-starter-stream-processor-httpclient/src/main/java/org/springframework/cloud/stream/app/httpclient/processor/HttpclientProcessorFunctionConfiguration.java]: Why `return null`? Why just not re-throw?
I know this is an existing code, but still...

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-cloud-starter-stream-processor-httpclient/src/main/java/org/springframework/cloud/stream/app/httpclient/processor/HttpclientProcessorFunctionConfiguration.java [hunk 2]
```diff
@@ -99,8 +94,7 @@ else if (properties.getBodyExpression() != null) {
 				return properties.getReplyExpression().getValue(response);
 			}
 			catch (Exception e) {
-				LOG.warn("Error in HTTP request", e);
-				return null;
+				throw new RuntimeException(e.getMessage(), e);
```
Comment (artembilan) [inline spring-cloud-starter-stream-processor-httpclient/src/main/java/org/springframework/cloud/stream/app/httpclient/processor/HttpclientProcessorFunctionConfiguration.java]: Why do we need to catch exceptions here at all?
Can't we just rely on the regular execution flow?

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-cloud-starter-stream-processor-httpclient/src/main/java/org/springframework/cloud/stream/app/httpclient/processor/HttpclientProcessorFunctionConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+			else {
+				body = message.getPayload();
+			}
+
+			URI uri;
+			try {
+				uri = new URI(url);
+			}
+			catch (URISyntaxException e) {
+				throw new RuntimeException(e.getMessage(), e);
```
Comment (artembilan) [inline spring-cloud-starter-stream-processor-httpclient/src/main/java/org/springframework/cloud/stream/app/httpclient/processor/HttpclientProcessorFunctionConfiguration.java]: `IllegalStateException` then? The `RuntimeException` is too generic and I don't know why it is not declared as `abstract`...

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-cloud-starter-stream-processor-httpclient/src/main/java/org/springframework/cloud/stream/app/httpclient/processor/HttpclientProcessorFunctionConfiguration.java (hunk 3 above)
Comment (dturanski) [reply spring-cloud-starter-stream-processor-httpclient/src/main/java/org/springframework/cloud/stream/app/httpclient/processor/HttpclientProcessorFunctionConfiguration.java]: I use RTE a lot in these situations. IlegalState or IlegalArg is appropriate

**Pull request merged**